command-expansion-server/src/lib/codegen/CodegenHelpers.ts

```diff
@@ -12,3 +12,5 @@ export function indent(text: string): string {
     .map(line => `\t${line}`)
     .join('\n');
 }
+
+export type Mutable<T> = { -readonly [P in keyof T]: T[P] };
```

Comment:
I have never seen a negative type before?
`-readonly`

Reply:
It's a cute little utility thing. Useful in cases like this where we really want it to be immutable, but we want to knowingly override that here

Reply:
https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#mapping-modifiers